examples/celeba.py

```diff
@@ -0,0 +1,183 @@
+import os
+import time
+
+import json
+import numpy as np
+import torch
+
+from appfl.config import *
+from appfl.misc.data import *
+from models.cnn import *
+import appfl.run_serial as rs
+import appfl.run_mpi as rm
+from mpi4py import MPI
+from models.utils import get_model
+import argparse
+
+from PIL import Image
+from PIL import ImageOps
+
+""" read arguments """
+
+parser = argparse.ArgumentParser()
+
+parser.add_argument("--device", type=str, default="cpu")
+
+## dataset
+parser.add_argument("--dataset", type=str, default="CELEBA")
+parser.add_argument("--num_channel", type=int, default=3)
+parser.add_argument("--num_classes", type=int, default=2)
+parser.add_argument("--num_pixel", type=int, default=218)
+parser.add_argument("--model", type=str, default="CNN")
+
+## clients
+parser.add_argument("--num_clients", type=int, default=1)
+parser.add_argument("--client_optimizer", type=str, default="Adam")
+parser.add_argument("--client_lr", type=float, default=1e-3)
+parser.add_argument("--num_local_epochs", type=int, default=1)
+
+## server
+parser.add_argument("--server", type=str, default="ServerFedAvg")
+parser.add_argument("--num_epochs", type=int, default=2)
+
+parser.add_argument("--server_lr", type=float, required=False)
+parser.add_argument("--mparam_1", type=float, required=False)
+parser.add_argument("--mparam_2", type=float, required=False)
+parser.add_argument("--adapt_param", type=float, required=False)
+
+parser.add_argument("--pretrained", type=int, default=0)
+
+args = parser.parse_args()
+
+if torch.cuda.is_available():
+    args.device = "cuda"
+    args.num_gpu = torch.cuda.device_count()
+
+
+dir = os.getcwd() + "/datasets/RawData/%s" % (args.dataset)
+
+def get_data(comm: MPI.Comm):
+    # test data for a server 
+    test_data_raw = {}
+    test_data_input = []
+    test_data_label = []
+    imagedir = dir + "/raw/img_align_celeba/"

```

Comment:
I think we should as shell script for downloading the dataset, or instructions on how to download it. Also, we need to update the ReadME.md file in the examples folder